api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValue.java

```diff
@@ -0,0 +1,161 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.spi;
+
+import java.util.Objects;
+
+/**
+ * The ConfigValue is a metadata object that holds additional information after the lookup of a configuration.
+ * <p>
+ *
+ * Right now, it is able to hold information like the configuration name, value, the Config Source from where
+ * the configuration was loaded, the ordinal of the Config Source and a line number from where the configuration was
+ * read if exists.
+ * <p>
+ *
+ * This is used together with {@link ConfigSource} to expose the Configuration lookup metadata.
+ *
+ * @author <a href="mailto:radcortez@yahoo.com">Roberto Cortez</a>
+ */
+public class ConfigValue {
+    private final String name;
+    private final String value;
+    private final String configSourceName;
+    private final int configSourceOrdinal;
+
+    private final int lineNumber;
+
+    private ConfigValue(final ConfigValueBuilder builder) {
+        this.name = builder.name;
+        this.value = builder.value;
+        this.configSourceName = builder.configSourceName;
+        this.configSourceOrdinal = builder.configSourceOrdinal;
+        this.lineNumber = builder.lineNumber;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public String getConfigSourceName() {
+        return configSourceName;
+    }
+
+    public int getConfigSourceOrdinal() {
+        return configSourceOrdinal;
+    }
+
+    public int getLineNumber() {
+        return lineNumber;
+    }
+
+    public ConfigValue withName(final String name) {
+        return from().withName(name).build();
+    }
+
+    public ConfigValue withValue(final String value) {
+        return from().withValue(value).build();
+    }
+
+    public ConfigValue withConfigSourceName(final String configSourceName) {
+        return from().withConfigSourceName(configSourceName).build();
+    }
+
+    public ConfigValue withConfigSourceOrdinal(final int configSourceOrdinal) {
+        return from().withConfigSourceOrdinal(configSourceOrdinal).build();
+    }
+
+    public ConfigValue withLineNumber(final int lineNumber) {
+        return from().withLineNumber(lineNumber).build();
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final ConfigValue that = (ConfigValue) o;
+        return name.equals(that.name) &&
+                value.equals(that.value) &&
+                configSourceName.equals(that.configSourceName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, value, configSourceName);
+    }
+
+    ConfigValueBuilder from() {
```

Comment:
How useful is this?

Reply:
Yes, maybe we need to recheck the visibly modifiers. I'll have a look